// deno.ns/dist/index.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    Object.defineProperty(o, k2, { enumerable: true, get: function() { return m[k]; } });
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
var __exportStar = (this && this.__exportStar) || function(m, exports) {
    for (var p in m) if (p !== "default" && !Object.prototype.hasOwnProperty.call(exports, p)) __createBinding(exports, m, p);
};
var _a, _b, _c, _d, _e, _f;
Object.defineProperty(exports, "__esModule", { value: true });
exports.Response = exports.Request = exports.Headers = exports.FormData = exports.File = exports.fetch = exports.Deno = exports.crypto = exports.Blob = void 0;
var buffer_1 = require("buffer");
Object.defineProperty(exports, "Blob", { enumerable: true, get: function () { return buffer_1.Blob; } });
var crypto_1 = require("crypto");
Object.defineProperty(exports, "crypto", { enumerable: true, get: function () { return crypto_1.webcrypto; } });
exports.Deno = __importStar(require("./deno.js"));
__exportStar(require("./util/mod.js"), exports);
// TODO: Wait for node builtin WhatWG Streams or fetch-blob to export them without assigning to globalThis
const undici = __importStar(require("undici"));
// fallback to using the global types and values if they exist
exports.fetch = (_a = globalThis["fetch"]) !== null && _a !== void 0 ? _a : undici.fetch;
exports.File = (_b = globalThis["File"]) !== null && _b !== void 0 ? _b : undici.File;
exports.FormData = (_c = globalThis["FormData"]) !== null && _c !== void 0 ? _c : undici.FormData;
exports.Headers = (_d = globalThis["Headers"]) !== null && _d !== void 0 ? _d : undici.Headers;
exports.Request = (_e = globalThis["Request"]) !== null && _e !== void 0 ? _e : undici.Request;
exports.Response = (_f = globalThis["Response"]) !== null && _f !== void 0 ? _f : undici.Response;
